Code/GraphMol/MolOps.cpp

```diff
@@ -1168,5 +1168,105 @@ std::vector<int> hapticBondEndpoints(const Bond *bond) {
   return oats;
 }
 }  // end of namespace details
-};  // end of namespace MolOps
-};  // end of namespace RDKit
+
+void expandAttachmentPoints(RWMol &mol, bool addAsQueries) {
+  for (auto atom : mol.atoms()) {
+    int value;
+    if (atom->getPropIfPresent(common_properties::molAttachPoint, value)) {
+      std::vector<int> tgtVals;
+      if (value == 1 || value == -1) {
+        tgtVals.push_back(1);
+      }
+      if (value == 2 || value == -1) {
+        tgtVals.push_back(2);
+      }
+      if (tgtVals.empty()) {
+        BOOST_LOG(rdWarningLog)
+            << "Invalid value for molAttachPoint: " << value << " on atom "
+            << atom->getIdx() << ". Not expanding this atttachment point."
+            << std::endl;
+        continue;
+      }
+      for (auto tval : tgtVals) {
+        atom->clearProp(common_properties::molAttachPoint);
+        Atom *newAtom = nullptr;
+        if (addAsQueries) {
+          newAtom = new QueryAtom(0);
+          newAtom->setQuery(RDKit::makeAtomNullQuery());
+        } else {
+          newAtom = new Atom(0);
+        }
+        newAtom->setProp(common_properties::_fromAttachPoint, tval);
+        bool updateLabel = false;
+        bool takeOwnership = true;
+        auto idx = mol.addAtom(newAtom, updateLabel, takeOwnership);
+        mol.addBond(atom->getIdx(), idx, Bond::SINGLE);
+      }
+    }
```

Comment:
The code we have does this:
```
auto explicit_h_count = atom->getNumExplicitHs();
<add attachment dummies>
if (explicit_h_count != 0) {
    atom->setNumExplicitHs(explicit_h_count - num_dummies);
}
```
should we also be updating the parent atom explicit Hs?

We also call `compute2DCoords` with all non-new-attachments frozen to make sure the new atoms get reasonable coordinates. Is that also valuable?

Reply:
> The code we have does this:
> 
> ```
> auto explicit_h_count = atom->getNumExplicitHs();
> <add attachment dummies>
> if (explicit_h_count != 0) {
>     atom->setNumExplicitHs(explicit_h_count - num_dummies);
> }
> ```
> 
> should we also be updating the parent atom explicit Hs?

Yes. Thanks.

> 
> We also call `compute2DCoords` with all non-new-attachments frozen to make sure the new atoms get reasonable coordinates. Is that also valuable?

Yes, I will add an `addCoords` option which behaves the same way as it does in `addHs()`